lib/Doctrine/DBAL/Driver/SQLite3/SQLite3Connection.php

```diff
@@ -0,0 +1,216 @@
+ */
+
+namespace Doctrine\DBAL\Driver\SQLite3;
+
+use Doctrine\DBAL\Driver\Connection;
+use Doctrine\DBAL\Driver\ServerInfoAwareConnection;
+
+/**
+ * SQLite3 implementation of the Connection interface.
+ *
+ * @since 2.6
+ * @author Ben Morel <benjamin.morel@gmail.com>
+ * @author Bill Schaller <bill@zeroedin.com>
+ */
+class SQLite3Connection extends SQLite3Abstract implements Connection, ServerInfoAwareConnection
+{
+    /**
+     * @param string      $filename
+     * @param int|null    $flags
+     * @param string|null $encryption_key
+     *
+     * @throws SQLite3Exception
+     */
+    public function __construct($filename, $flags = null, $encryption_key = null)
+    {
+        try {
+            if ($flags === null) {
+                $this->sqlite3 = new \SQLite3($filename);
+            } elseif ($encryption_key === null) {
+                $this->sqlite3 = new \SQLite3($filename, $flags);
+            } else {
+                $this->sqlite3 = new \SQLite3($filename, $flags, $encryption_key);
+            }
+
+            $this->sqlite3->enableExceptions(true);
+        } catch (\Exception $e) {
+            throw $this->wrapDriverException($e);
+        }
+    }
+
+    /**
+     * Returns the underlying SQLite3 object.
+     *
+     * @return \SQLite3
+     */
+    public function getSQLite3()
+    {
+        return $this->sqlite3;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function prepare($prepareString)
+    {
+        try {
+            $statement = @ $this->sqlite3->prepare($prepareString);
+        } catch (\Exception $e) {
+            throw $this->wrapDriverException($e);
+        }
+
+        $this->throwExceptionOnError();
+
+        return new SQLite3Statement($this->sqlite3, $statement);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function query()
+    {
+        $query = func_get_arg(0);
+
+        $statement = $this->prepare($query);
+        $statement->execute();
+
+        return $statement;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function quote($input, $type = \PDO::PARAM_STR)
+    {
+        return "'" . \SQLite3::escapeString($input) . "'";
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exec($statement)
+    {
+        try {
+            $this->sqlite3->exec($statement);
+        } catch (\Exception $e) {
+            throw $this->wrapDriverException($e);
+        }
+
+        $this->throwExceptionOnError();
+
+        return $this->sqlite3->changes();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function lastInsertId($name = null)
+    {
+        try {
+            return $this->sqlite3->lastInsertRowID();
+        } catch (\Exception $e) {
+            throw $this->wrapDriverException($e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function beginTransaction()
+    {
+        try {
+            $this->sqlite3->exec('BEGIN');
+        } catch (\Exception $e) {
```

Comment:
Fully agree @zeroedin-bill. The only problem is we have contracts for those kind of things which we *SHOULD* not violate. Otherwise having contracts at all is rather useless. On the other hand as you mentioned, I guess if even ORM is relying on the false behaviour, fixing all the drivers to adhere to the contract will probably break more things than it solves.
I really have no idea which direction to go :> maybe @Ocramius has an opinion here...